collector/self-checks.cpp

```diff
@@ -0,0 +1,124 @@
+
+#include <cstdio>
+#include <cstdlib>
+#include <iostream>
+#include <unistd.h>
+
+#include <netinet/in.h>
+#include <sys/errno.h>
+#include <sys/select.h>
+#include <sys/wait.h>
+
+const int g_connection_retries = 5;
+const int g_timeout_seconds = 10;
+const int g_sleep_seconds = 1;
+
+/**
+ * @brief starts a listening socket for the given port, and waits for a connection
+ *        for a short time. If nothing connects in time, the socket is closed.
+ */
+bool startListeningPort(uint16_t port) {
+  bool result = false;
+  struct sockaddr_in sock_addr = {};
+  int connection = -1;
+  fd_set sockets;
+  struct timeval timeout = {.tv_sec = g_timeout_seconds};
+
+  int listener = socket(AF_INET, SOCK_STREAM, 0);
+
+  if (listener < 0) {
+    return false;
```

Comment:
In `connectToPort` if failed to open a socket, it will be explicitly closed. Here it will return false, does it have to be closed as well?

Reply:
I'll make this do the same, for consistency